dimod/binary_quadratic_model.py

```diff
@@ -194,20 +202,26 @@ class for the K4 complete graph, where the nodes have biases
     BINARY = Vartype.BINARY
 
     @vartype_argument('vartype')
-    def __init__(self, linear, quadratic, offset, vartype, **kwargs):
+    def __init__(self, linear, quadratic, offset, vartype, dtype=object, **kwargs):
         self.linear = {}
         self.quadratic = {}
         self.adj = {}
-        self.offset = offset  # we are agnostic to type, though generally should behave like a number
+
+        if dtype is object:
+            self.offset = offset  # we are agnostic to type, though generally should behave like a number
+        else:
+            self.offset = dtype(offset)
```

Comment:
Is self.offset an actual number (the energy offset) or a data type?

Reply:
An actual number.